include/groonga/thread.h

```diff
@@ -93,7 +93,49 @@ typedef uint32_t (*grn_thread_get_limit_func)(void *data);
  */
 GRN_API void
 grn_thread_set_get_limit_func(grn_thread_get_limit_func func, void *data);
+/**
+ * \brief The type of function that sets the max number of threads.
+ *
+ * \since 5.0.7
+ *
+ * This is the function pointer type for custom functions that set the
+ * maximum number of threads. When set via \ref grn_thread_set_set_limit_func,
+ * this function will be called by \ref grn_thread_set_limit.
+ *
+ * \param new_limit The new max number of threads to set.
+ * \param data User data passed when the function was registered.
+ */
 typedef void (*grn_thread_set_limit_func)(uint32_t new_limit, void *data);
+/**
+ * \brief Set the custom function that sets the max number of threads.
+ *
+ * \since 5.0.7
+ *
+ * This function sets a custom function that will be called when
+ * \ref grn_thread_set_limit is invoked. The custom function allows
+ * applications to handle thread limit changes.
+ *
+ * \p data is passed to \p func when \p func is called from
+ * \ref grn_thread_set_limit.
+ *
+ * For example usage:
+ * ```c
+ * static uint32_t max_n_threads;
```

Comment:
```suggestion
 * static uint32_t n_max_threads;
```

Reply:
We use the `max_n_threads` in our code base. So I chose this name here.  
https://github.com/groonga/groonga/blob/d3469e9119dea22bdff536691297a5a23f9be8c5/src/groonga.c#L3892

Reply:
I see.